[FEAT] 개인회원 CRU

smallsnail-study opened the pull request:
<!-- Assignees는 본인, Reviewrs는 팀원 선택 -->

## #️⃣연관된 이슈

> #7 

## 📝작업 내용

> 개인회원 CRU

>  어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

> User 엔티티에 Role Enum String으로 수정했습니다. DB에서 구분하기 쉽도록 하기 위함이에요! DB 반영부탁드려요

## 체크리스트
- [x] Reviewrs를 설정했습니다

smallsnail-study committed:
feat: 개인회원 주소수정 메서드 추가
AddressService.java
- 개인회원 정보수정시 사용하는 주소수정 메서드 추가

smallsnail-study committed:
feat: 개인회원 정보수정 메서드 추가
Individual.java
- 닉네임, 비밀번호, 주소 수정가능
- 프로필이미지 수정 추후 반영 예정

smallsnail-study committed:
feat: 개인회원 조회관련 repository 별도분리
IndividualRepository.java
- 조회관련 userRepository 사용 시 문제발생으로 IndividualRepository를 별도로 구현

smallsnail-study committed:
feat: 개인회원가입용 Dto 생성
RequestIndividualJoinDto.java
- 회원가입 시 입력할 항목들 dto로 생성

smallsnail-study committed:
feat: 개인회원 정보수정용 Dto 생성
RequestIndividualUpdateDto.java
- 개인회원 정보 수정 시 필요항목들 Dto

smallsnail-study committed:
feat: ResponseusetDto에 개인회원 추가
ResponseUserDto.java
- 조회 시 사용할 dto

smallsnail-study committed:
feat: UserEntity 수정(Enum설정추가,email인증여부추가,권한관련추가)
User.java
- EnumType.STRING 추가
- email인증여부 주석으로 추가 -> 구현 시 수정예정
- 권한관련 추가

smallsnail-study committed:
feat: UserService 개인회원(CRU)추가
UserService.java
- 개인회원 가입
- 개인회원 정보수정
- 개인회원 전체조회
- 개인회원 단일조회

smallsnail-study committed:
feat: UserRepository findByEmail 공통으로 수정
UserRepository.java
- findByEmail 공통 User로 수정

smallsnail-study committed:
feat: UserController 개인회원 CRU 추가
UserController.java
- 개인회원가입
- 개인회원 정보수정
- 개인회원 전체조회
- 개인회원 단일조회

Jyoungjo reviewed (approved):
:)

havegrit commented on BE/src/main/java/com/myapp/warmwave/domain/address/service/AddressService.java:
```diff
@@ -38,4 +40,12 @@ public void updateAddress(RequestInstitutionUpdateDto dto, Institution instituti
+        originalAddress.update(dto.getFullAddr(), dto.getSdName(), dto.getSggName(), dto.getDetails());
+
+        return originalAddress;
+    }
```
update 작업`@Transactional` 어노테이션 작성해주셔야할 것 같습니다 !

그리고 entity 내용 업데이트할 때, spring data jpa가 변경내용 알아서 저장해주지만, spring data jpa에 의존적인 코드 작성을 피하기 위해 save() 직접 호출해서 저장하는 방식으로 갔으면 어떨까 하는 생각입니다.

smallsnail-study replied:
수정완료 했습니다

smallsnail-study committed:
feat: AddressService udpate로직 수정
AddressService.java
- @Transactional 추가
- updateIndividualAddress addressRepository.save로 수정

smallsnail-study committed:
Merge branch 'develop' into feature/#7
